core/glrender/src/main/java/edu/cmu/cs/dennisc/render/gl/imp/adapters/GlrBox.java

```diff
@@ -55,266 +55,195 @@
+                direction.y = 1;
+                break;
+            case 4:
+                origin.z = this.zMin;
+                direction.z = -1;
+                break;
+            case 5:
+                origin.z = this.zMax;
+                direction.z = 1;
+                break;
+            default:
+                rv.setNaN();
+                return rv;
+        }
+        GlrGeometry.getIntersectionInSourceFromPlaneInLocal(rv, ray, m, origin.x, origin.y, origin.z, direction.x, direction.y, direction.z);
+        return rv;
+    }
+
+    @Override
+    protected void propertyChanged(InstanceProperty<?> property) {
+        if (property == owner.xMinimum) {
+            this.xMin = owner.xMinimum.getValue();
+            setIsGeometryChanged(true);
+        } else if (property == owner.xMaximum) {
+            this.xMax = owner.xMaximum.getValue();
+            setIsGeometryChanged(true);
+        } else if (property == owner.yMinimum) {
+            this.yMin = owner.yMinimum.getValue();
+            setIsGeometryChanged(true);
+        } else if (property == owner.yMaximum) {
+            this.yMax = owner.yMaximum.getValue();
+            setIsGeometryChanged(true);
+        } else if (property == owner.zMinimum) {
+            this.zMin = owner.zMinimum.getValue();
+            setIsGeometryChanged(true);
+        } else if (property == owner.zMaximum) {
+            this.zMax = owner.zMaximum.getValue();
+            setIsGeometryChanged(true);
+        } else {
+            super.propertyChanged(property);
+        }
+    }
+
+    private double xMin;
+    private double xMax;
+    private double yMin;
+    private double yMax;
+    private double zMin;
+    private double zMax;
+    private static final int numberOfCorner = 4;
+    private static final int[] leftNormal3d = {-1, 0, 0};
+    private static final int[] rightNormal3d = {1, 0, 0};
+    private static final int[] bottomNormal3d = {0, -1, 0};
+    private static final int[] topNormal3d = {0, 1, 0};
+    private static final int[] frontNormal3d = {0, 0, -1};
+    private static final int[] backNormal3d = {0, 0, 1};
+    private static final float[][] leftTexCoord = {{.75f, .66f}, {.75f, .33f}, {.5f, .33f}, {.5f, .66f}};
+    private static final float[][] rightTexCoord = {{.25f, .66f}, {.25f, .33f}, {.0f, .33f}, {.0f, .66f}};
+    private static final float[][] bottomTexCoord = {{.5f, .66f}, {.25f, .66f}, {.25f, 1.0f}, {.5f, 1.0f}};
+    private static final float[][] topTexCoord = {{.5f, .0f}, {.25f, .0f}, {.25f, .33f}, {.5f, .33f}};
+    private static final float[][] frontTexCoord = {{.5f, .33f}, {.25f, .33f}, {.25f, .66f}, {.5f, .66f}};
+    private static final float[][] backTexCoord = {{.75f, .66f}, {1.0f, .66f}, {1.0f, .33f}, {.75f, .33f}};
+    private static Map<String, int[]> normal3dMap = new HashMap<>();
+    private static Map<String, float[][]> texCoordMap = new HashMap<>();
+    private static Map<String, double[][]> vertexMap = new HashMap<>();
```

Comment:
This being `static` means the same map is used for all boxes in the scene.
It probably should not be.

In theory it could be possible for multiple cubes to overwrite each and interfere with rendering.